### ISSUE TITLE
Fails to Install

### ISSUE THREAD

**Issue by Sovjetskii** [filmackay/blpapi-py#3]
When trying to build (python setup.py build --compiler msvc) using a 3.4.3 (32-bit and 64-bit), I receive the following errors:

```
C:\Users\<user>\Desktop\blpapi3>python setup.py build
running build
running build_py
running build_ext
building 'blpapi._internals' extension
C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\Bin\amd64\cl.exe /c /nolo
go /Ox /MD /W3 /GS- /DNDEBUG -IC:\blp\API\APIv3\C++API\v3.8.13.1\include -IC:\Py
thon34\include -IC:\Python34\include /Tpblpapi/internals_wrap.cxx /Fobuild\temp.
win32-3.4\Release\blpapi/internals_wrap.obj
internals_wrap.cxx
C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\INCLUDE\xlocale(323) : wa
rning C4530: C++ exception handler used, but unwind semantics are not enabled. S
pecify /EHsc
blpapi/internals_wrap.cxx(3921) : warning C4244: 'argument' : conversion from 'b
lpapi_Float64_t' to 'blpapi_Float32_t', possible loss of data
blpapi/internals_wrap.cxx(3935) : warning C4244: 'argument' : conversion from 'b
lpapi_Float64_t' to 'blpapi_Float32_t', possible loss of data
blpapi/internals_wrap.cxx(3954) : warning C4244: 'argument' : conversion from 'b
lpapi_Float64_t' to 'blpapi_Float32_t', possible loss of data
blpapi/internals_wrap.cxx(4008) : warning C4244: 'argument' : conversion from 'b
lpapi_Float64_t' to 'blpapi_Float32_t', possible loss of data
blpapi/internals_wrap.cxx(4025) : warning C4244: 'argument' : conversion from 'b
lpapi_Float64_t' to 'blpapi_Float32_t', possible loss of data
blpapi/internals_wrap.cxx(16923) : warning C4244: '=' : conversion from 'Py_ssiz
e_t' to 'unsigned int', possible loss of data
C:\Program Files (x86)\Microsoft Visual Studio 10.0\VC\Bin\amd64\link.exe /DLL /
nologo /INCREMENTAL:NO /LIBPATH:C:\blp\API\APIv3\C++API\v3.8.13.1\lib /LIBPATH:C
:\Python34\libs /LIBPATH:C:\Python34\PCbuild blpapi3_32.lib /EXPORT:PyInit__inte
rnals build\temp.win32-3.4\Release\blpapi/internals_wrap.obj /OUT:build\lib.win3
2-3.4\blpapi\_internals.pyd /IMPLIB:build\temp.win32-3.4\Release\blpapi\_interna
ls.lib /MANIFESTFILE:build\temp.win32-3.4\Release\blpapi\_internals.pyd.manifest
 /MANIFEST
internals_wrap.obj : warning LNK4197: export 'PyInit__internals' specified multi
ple times; using first specification
   Creating library build\temp.win32-3.4\Release\blpapi\_internals.lib and objec
t build\temp.win32-3.4\Release\blpapi\_internals.exp
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyBytes_AsS
tringAndSize referenced in function "char * __cdecl SWIG_Python_str_AsChar(struc
t _object *)" (?SWIG_Python_str_AsChar@@YAPEADPEAU_object@@@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyUnicode_A
sUTF8String referenced in function "char * __cdecl SWIG_Python_str_AsChar(struct
 _object *)" (?SWIG_Python_str_AsChar@@YAPEADPEAU_object@@@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyUnicode_F
romString referenced in function "struct _object * __cdecl SWIG_Python_str_FromC
har(char const *)" (?SWIG_Python_str_FromChar@@YAPEAU_object@@PEBD@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_Runti
meError referenced in function "struct _object * __cdecl SWIG_Python_ErrorType(i
nt)" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_Attri
buteError referenced in function "struct _object * __cdecl SWIG_Python_ErrorType
(int)" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_Syste
mError referenced in function "struct _object * __cdecl SWIG_Python_ErrorType(in
t)" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_Value
Error referenced in function "struct _object * __cdecl SWIG_Python_ErrorType(int
)" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_Synta
xError referenced in function "struct _object * __cdecl SWIG_Python_ErrorType(in
t)" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_Overf
lowError referenced in function "struct _object * __cdecl SWIG_Python_ErrorType(
int)" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_ZeroD
ivisionError referenced in function "struct _object * __cdecl SWIG_Python_ErrorT
ype(int)" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_TypeE
rror referenced in function "struct _object * __cdecl SWIG_Python_ErrorType(int)
" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_Index
Error referenced in function "struct _object * __cdecl SWIG_Python_ErrorType(int
)" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_IOErr
or referenced in function "struct _object * __cdecl SWIG_Python_ErrorType(int)"
(?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyExc_Memor
yError referenced in function "struct _object * __cdecl SWIG_Python_ErrorType(in
t)" (?SWIG_Python_ErrorType@@YAPEAU_object@@H@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyGILState_
Release referenced in function "void __cdecl SWIG_Python_SetErrorMsg(struct _obj
ect *,char const *)" (?SWIG_Python_SetErrorMsg@@YAXPEAU_object@@PEBD@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyGILState_
Ensure referenced in function "void __cdecl SWIG_Python_SetErrorMsg(struct _obje
ct *,char const *)" (?SWIG_Python_SetErrorMsg@@YAXPEAU_object@@PEBD@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyEval_Rest
oreThread referenced in function "struct blpapi_CorrelationId_t_ * __cdecl Corre
lationId_t_createEmpty(void)" (?CorrelationId_t_createEmpty@@YAPEAUblpapi_Correl
ationId_t_@@XZ)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyEval_Save
Thread referenced in function "struct blpapi_CorrelationId_t_ * __cdecl Correlat
ionId_t_createEmpty(void)" (?CorrelationId_t_createEmpty@@YAPEAUblpapi_Correlati
onId_t_@@XZ)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyInstanceM
ethod_New referenced in function SWIG_PyInstanceMethod_New
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyErr_SetSt
ring referenced in function "void __cdecl SWIG_Python_SetErrorMsg(struct _object
 *,char const *)" (?SWIG_Python_SetErrorMsg@@YAXPEAU_object@@PEBD@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyDict_SetI
temString referenced in function "void __cdecl SWIG_Python_SetConstant(struct _o
bject *,char const *,struct _object *)" (?SWIG_Python_SetConstant@@YAXPEAU_objec
t@@PEBD0@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyList_Appe
nd referenced in function "struct _object * __cdecl SWIG_Python_AppendOutput(str
uct _object *,struct _object *)" (?SWIG_Python_AppendOutput@@YAPEAU_object@@PEAU
1@0@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyList_SetI
tem referenced in function "struct _object * __cdecl SWIG_Python_AppendOutput(st
ruct _object *,struct _object *)" (?SWIG_Python_AppendOutput@@YAPEAU_object@@PEA
U1@0@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyList_New
referenced in function "struct _object * __cdecl SWIG_Python_AppendOutput(struct
 _object *,struct _object *)" (?SWIG_Python_AppendOutput@@YAPEAU_object@@PEAU1@0
@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp__Py_NoneStr
uct referenced in function "struct _object * __cdecl SWIG_Python_AppendOutput(st
ruct _object *,struct _object *)" (?SWIG_Python_AppendOutput@@YAPEAU_object@@PEA
U1@0@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_Py_BuildVal
ue referenced in function _SWIG_Py_None
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyErr_Clear
 referenced in function SwigPyClientData_New
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyErr_Occur
red referenced in function SwigPyClientData_New
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyTuple_Set
Item referenced in function SwigPyClientData_New
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyTuple_New
 referenced in function SwigPyClientData_New
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyObject_Ge
tAttrString referenced in function SwigPyClientData_New
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyObject_Is
Instance referenced in function SwigPyClientData_New
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyType_Type
 referenced in function SwigPyClientData_New
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyLong_From
VoidPtr referenced in function SwigPyObject_long
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_Py_DecRef r
eferenced in function SwigPyObject_repr
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyUnicode_C
oncat referenced in function SwigPyObject_repr
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyUnicode_F
romFormat referenced in function SwigPyObject_repr
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyBool_From
Long referenced in function SwigPyObject_richcompare
internals_wrap.obj : error LNK2019: unresolved external symbol __imp__Py_NotImpl
ementedStruct referenced in function SwigPyObject_richcompare
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyObject_Fr
ee referenced in function SwigPyObject_dealloc
...
age_correlationId referenced in function _wrap_blpapi_Message_correlationId
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Mess
age_elements referenced in function _wrap_blpapi_Message_elements
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Mess
age_fragmentType referenced in function _wrap_blpapi_Message_fragmentType
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Mess
age_addRef referenced in function _wrap_blpapi_Message_addRef
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Mess
age_release referenced in function _wrap_blpapi_Message_release
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Even
t_eventType referenced in function _wrap_blpapi_Event_eventType
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Even
t_release referenced in function _wrap_blpapi_Event_release
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Even
tQueue_create referenced in function _wrap_blpapi_EventQueue_create
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Even
tQueue_destroy referenced in function _wrap_blpapi_EventQueue_destroy
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Even
tQueue_nextEvent referenced in function _wrap_blpapi_EventQueue_nextEvent
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Even
tQueue_purge referenced in function _wrap_blpapi_EventQueue_purge
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Even
tQueue_tryNextEvent referenced in function _wrap_blpapi_EventQueue_tryNextEvent
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Mess
ageIterator_create referenced in function _wrap_blpapi_MessageIterator_create
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Mess
ageIterator_destroy referenced in function _wrap_blpapi_MessageIterator_destroy
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Mess
ageIterator_next referenced in function _wrap_blpapi_MessageIterator_next
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Iden
tity_release referenced in function _wrap_blpapi_Identity_release
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Iden
tity_addRef referenced in function _wrap_blpapi_Identity_addRef
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Iden
tity_hasEntitlements referenced in function _wrap_blpapi_Identity_hasEntitlement
s
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Iden
tity_isAuthorized referenced in function _wrap_blpapi_Identity_isAuthorized
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Iden
tity_getSeatType referenced in function _wrap_blpapi_Identity_getSeatType
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Abst
ractSession_cancel referenced in function _wrap_blpapi_AbstractSession_cancel
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Abst
ractSession_sendAuthorizationRequest referenced in function _wrap_blpapi_Abstrac
tSession_sendAuthorizationRequest
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Abst
ractSession_openService referenced in function _wrap_blpapi_AbstractSession_open
Service
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Abst
ractSession_openServiceAsync referenced in function _wrap_blpapi_AbstractSession
_openServiceAsync
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Abst
ractSession_generateToken referenced in function _wrap_blpapi_AbstractSession_ge
nerateToken
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Abst
ractSession_getService referenced in function _wrap_blpapi_AbstractSession_getSe
rvice
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Abst
ractSession_createIdentity referenced in function _wrap_blpapi_AbstractSession_c
reateIdentity
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_start referenced in function _wrap_blpapi_Session_start
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_startAsync referenced in function _wrap_blpapi_Session_startAsync
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_stop referenced in function _wrap_blpapi_Session_stop
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_stopAsync referenced in function _wrap_blpapi_Session_stopAsync
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_nextEvent referenced in function _wrap_blpapi_Session_nextEvent
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_tryNextEvent referenced in function _wrap_blpapi_Session_tryNextEvent
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_subscribe referenced in function _wrap_blpapi_Session_subscribe
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_resubscribe referenced in function _wrap_blpapi_Session_resubscribe
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_resubscribeWithId referenced in function _wrap_blpapi_Session_resubscribeWit
hId
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_unsubscribe referenced in function _wrap_blpapi_Session_unsubscribe
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_setStatusCorrelationId referenced in function _wrap_blpapi_Session_setStatus
CorrelationId
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_sendRequest referenced in function _wrap_blpapi_Session_sendRequest
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sess
ion_getAbstractSession referenced in function _wrap_blpapi_Session_getAbstractSe
ssion
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_extractAttributeFromResolutionSuccess referenced in function _wrap_bl
papi_ResolutionList_extractAttributeFromResolutionSuccess
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_create referenced in function _wrap_blpapi_ResolutionList_create
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_destroy referenced in function _wrap_blpapi_ResolutionList_destroy
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_add referenced in function _wrap_blpapi_ResolutionList_add
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_addFromMessage referenced in function _wrap_blpapi_ResolutionList_add
FromMessage
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_addAttribute referenced in function _wrap_blpapi_ResolutionList_addAt
tribute
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_correlationIdAt referenced in function _wrap_blpapi_ResolutionList_co
rrelationIdAt
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_topicString referenced in function _wrap_blpapi_ResolutionList_topicS
tring
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_topicStringAt referenced in function _wrap_blpapi_ResolutionList_topi
cStringAt
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_status referenced in function _wrap_blpapi_ResolutionList_status
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_statusAt referenced in function _wrap_blpapi_ResolutionList_statusAt
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_attribute referenced in function _wrap_blpapi_ResolutionList_attribut
e
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_attributeAt referenced in function _wrap_blpapi_ResolutionList_attrib
uteAt
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_message referenced in function _wrap_blpapi_ResolutionList_message
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_messageAt referenced in function _wrap_blpapi_ResolutionList_messageA
t
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Reso
lutionList_size referenced in function _wrap_blpapi_ResolutionList_size
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
c_create referenced in function _wrap_blpapi_Topic_create
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
c_destroy referenced in function _wrap_blpapi_Topic_destroy
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
c_compare referenced in function _wrap_blpapi_Topic_compare
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
c_service referenced in function _wrap_blpapi_Topic_service
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
c_isActive referenced in function _wrap_blpapi_Topic_isActive
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_destroy referenced in function _wrap_blpapi_TopicList_destroy
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_add referenced in function _wrap_blpapi_TopicList_add
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_addFromMessage referenced in function _wrap_blpapi_TopicList_addFromMessag
e
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_correlationIdAt referenced in function _wrap_blpapi_TopicList_correlationI
dAt
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_topicString referenced in function _wrap_blpapi_TopicList_topicString
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_topicStringAt referenced in function _wrap_blpapi_TopicList_topicStringAt
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_status referenced in function _wrap_blpapi_TopicList_status
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_statusAt referenced in function _wrap_blpapi_TopicList_statusAt
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_message referenced in function _wrap_blpapi_TopicList_message
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_messageAt referenced in function _wrap_blpapi_TopicList_messageAt
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Topi
cList_size referenced in function _wrap_blpapi_TopicList_size
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_start referenced in function _wrap_blpapi_ProviderSession_start
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_startAsync referenced in function _wrap_blpapi_ProviderSession_start
Async
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_stop referenced in function _wrap_blpapi_ProviderSession_stop
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_stopAsync referenced in function _wrap_blpapi_ProviderSession_stopAs
ync
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_nextEvent referenced in function _wrap_blpapi_ProviderSession_nextEv
ent
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_tryNextEvent referenced in function _wrap_blpapi_ProviderSession_try
NextEvent
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_registerService referenced in function _wrap_blpapi_ProviderSession_
registerService
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_activateSubServiceCodeRange referenced in function _wrap_blpapi_Prov
iderSession_activateSubServiceCodeRange
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_deactivateSubServiceCodeRange referenced in function _wrap_blpapi_Pr
oviderSession_deactivateSubServiceCodeRange
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_registerServiceAsync referenced in function _wrap_blpapi_ProviderSes
sion_registerServiceAsync
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_deregisterService referenced in function _wrap_blpapi_ProviderSessio
n_deregisterService
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_resolve referenced in function _wrap_blpapi_ProviderSession_resolve
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_resolveAsync referenced in function _wrap_blpapi_ProviderSession_res
olveAsync
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_createTopics referenced in function _wrap_blpapi_ProviderSession_cre
ateTopics
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_createTopicsAsync referenced in function _wrap_blpapi_ProviderSessio
n_createTopicsAsync
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_getTopic referenced in function _wrap_blpapi_ProviderSession_getTopi
c
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_createTopic referenced in function _wrap_blpapi_ProviderSession_crea
teTopic
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_createServiceStatusTopic referenced in function _wrap_blpapi_Provide
rSession_createServiceStatusTopic
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_deleteTopics referenced in function _wrap_blpapi_ProviderSession_del
eteTopics
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_publish referenced in function _wrap_blpapi_ProviderSession_publish
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_sendResponse referenced in function _wrap_blpapi_ProviderSession_sen
dResponse
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Prov
iderSession_getAbstractSession referenced in function _wrap_blpapi_ProviderSessi
on_getAbstractSession
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_create referenced in function _wrap_blpapi_ServiceRegistr
ationOptions_create
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_duplicate referenced in function _wrap_blpapi_ServiceRegi
strationOptions_duplicate
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_destroy referenced in function _wrap_blpapi_ServiceRegist
rationOptions_destroy
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_copy referenced in function _wrap_blpapi_ServiceRegistrat
ionOptions_copy
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_addActiveSubServiceCodeRange referenced in function _wrap
_blpapi_ServiceRegistrationOptions_addActiveSubServiceCodeRange
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_removeAllActiveSubServiceCodeRanges referenced in functio
n _wrap_blpapi_ServiceRegistrationOptions_removeAllActiveSubServiceCodeRanges
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_setGroupId referenced in function _wrap_blpapi_ServiceReg
istrationOptions_setGroupId
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyBytes_Siz
e referenced in function _wrap_blpapi_ServiceRegistrationOptions_setGroupId
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyBytes_AsS
tring referenced in function _wrap_blpapi_ServiceRegistrationOptions_setGroupId
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_setServicePriority referenced in function _wrap_blpapi_Se
rviceRegistrationOptions_setServicePriority
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_setPartsToRegister referenced in function _wrap_blpapi_Se
rviceRegistrationOptions_setPartsToRegister
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_getGroupId referenced in function _wrap_blpapi_ServiceReg
istrationOptions_getGroupId
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_getServicePriority referenced in function _wrap_blpapi_Se
rviceRegistrationOptions_getServicePriority
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
iceRegistrationOptions_getPartsToRegister referenced in function _wrap_blpapi_Se
rviceRegistrationOptions_getPartsToRegister
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyEval_Init
Threads referenced in function PyInit__internals
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyModule_Ge
tDict referenced in function PyInit__internals
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_PyModule_Cr
eate2 referenced in function PyInit__internals
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Serv
ice_print referenced in function "class std::basic_string<char,struct std::char_
traits<char>,class std::allocator<char> > __cdecl blpapi_Service_printHelper(str
uct blpapi_Service *,int,int)" (?blpapi_Service_printHelper@@YA?AV?$basic_string
@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@PEAUblpapi_Service@@HH@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sche
maElementDefinition_print referenced in function "class std::basic_string<char,s
truct std::char_traits<char>,class std::allocator<char> > __cdecl blpapi_SchemaE
lementDefinition_printHelper(void * *,int,int)" (?blpapi_SchemaElementDefinition
_printHelper@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@
PEAPEAXHH@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Sche
maTypeDefinition_print referenced in function "class std::basic_string<char,stru
ct std::char_traits<char>,class std::allocator<char> > __cdecl blpapi_SchemaType
Definition_printHelper(void * *,int,int)" (?blpapi_SchemaTypeDefinition_printHel
per@@YA?AV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@PEAPEAXHH
@Z)
internals_wrap.obj : error LNK2019: unresolved external symbol __imp_blpapi_Elem
ent_print referenced in function "class std::basic_string<char,struct std::char_
traits<char>,class std::allocator<char> > __cdecl blpapi_Element_printHelper(str
uct blpapi_Element *,int,int)" (?blpapi_Element_printHelper@@YA?AV?$basic_string
@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@PEAUblpapi_Element@@HH@Z)
build\lib.win32-3.4\blpapi\_internals.pyd : fatal error LNK1120: 391 unresolved
```
I've tried building the standard BLP API (v.3.5.5) with no errors (using a 32-bit 2.7 interpreter). 

You will note that the build call does not include (--compiler msvc), but I previously ran build with the argument as well - no difference.

Thanks very much in advance! I would really hate to revert to 2.7 for the rest of my project.

**Comment by Sovjetskii** [comment]
To be clear, I previously ran build with the --compiler msvc argument as well - no difference.

**Comment by Sovjetskii** [comment]
Nevermind, figured it out. Thanks again for all your work here.

**Comment by filmackay** [comment]
Sorry mate, I missed this - wasn't ignoring. Can you post the resolution to the above?

**Comment by Sovjetskii** [comment]
It was due to my lack of diligence with regards to the environment I ran the compiler in. In the provided .bat file, you setenv to x64, so I assumed to run the 32-bit version, I would just exclude the call to setenv x64. However, what I didn't consider was that my native Windows SDK console was already defaulting to x64, so I actually had to set setenv /x86 /release and compile using the 32-bit MSVC.

I don't know to what extent such a conditional is possible, but it may be worth an if statement to return a contextual error: "your call to the msvc compiler and version of Python do not match".

Again, thanks for all your help with this port - it's a lifesaver.
